### ISSUE TITLE
t2 update: autocheck for out of date firmware with each cli usage

### ISSUE THREAD

**Issue by johnnyman727** [tessel/t2-cli#257]
With the Tessel 1 CLI, we would check if the CLI or firmware was out of date each time any command was run. This helped users know right away when an update was available rather than running it only when they run into problems.

**Comment by wprater** [comment]
maybe add a ~/.tessel/update_check with a datestamp as well, so it does not check _every_ time.

**Comment by rwaldron** [comment]
I don't think this should be done on every command. It would be more reasonable to: 

- check when the Tessel is powered on and gets a LAN connection 
- check when the user actually requests a check

**Comment by wprater** [comment]
I like the idea of showing the user when a new version is available, as other CLI tools do.  And agree that it should not be every command, which is why I suggested a date stamp file to ensure it's only called once per day (for example).  

or are you advocating you can only check for a new version be `t2 update -l` etc?

**Comment by LinusU** [comment]
There is a package on npm which many cli tool uses which actually fires away the update check in a background thread and is very smart on how it does things. Give me a minute...

**Comment by LinusU** [comment]
Here we go! :roller_coaster: 

https://github.com/yeoman/update-notifier

**Comment by johnnyman727** [comment]
Ah yes! That node module would be perfect. It's super handy to be notified when some component of the system is out of date.

**Comment by johnnyman727** [comment]
This is still a useful feature for anyone interested in contributing. The fix would entail adding a "minCLI" and "maxCLI" to each property of the `builds.json` file living on AWS (someone from the Tessel Team Members can give you access), then making use the package linked up to ensure the version of firmware on the Tessel (can be returned with `tessel.version`) and the CLI are compatible.

**Comment by Frijol** [comment]
Revisiting this issue, it looks like discussion was around update notifications for both CLI and firmware.
 
* [x] t2-cli notifies users of updates using the yeoman update notifier, but
* [ ] t2-firmware does not yet have an update notifier.

@johnnyman727 I'm trying to understand your proposed solution. It seems to have more to do with compatibility between CLI version and firmware version than to do with whether your firmware is up to date. Is that correct?
